src/InterventionBackend.php

```diff
@@ -567,135 +475,95 @@ function (InterventionImage $resource) use ($width, $height) {
     }
 
     /**
-     * Resize the image by preserving aspect ratio. By default, it will keep the image inside the maxWidth
-     * and maxHeight. Passing useAsMinimum will make the smaller dimension equal to the maximum corresponding dimension
-     *
-     * @param int $width
-     * @param int $height
-     * @param bool $useAsMinimum If true, image will be sized outside of these dimensions.
-     * If false (default) image will be sized inside these dimensions.
-     * @return static
+     * @inheritDoc
      */
-    public function resizeRatio($width, $height, $useAsMinimum = false)
+    public function resizeRatio(int $width, int $height, bool $useAsMinimum = false): static
     {
         return $this->createCloneWithResource(
             function (InterventionImage $resource) use ($width, $height, $useAsMinimum) {
-                return $resource->resize(
-                    $width,
-                    $height,
-                    function (Constraint $constraint) use ($useAsMinimum) {
-                        $constraint->aspectRatio();
-                        if (!$useAsMinimum) {
-                            $constraint->upsize();
-                        }
-                    }
-                );
+                if ($useAsMinimum) {
+                    return $resource->scale($width, $height);
+                }
+                return $resource->scaleDown($width, $height);
```

Comment:
If `$useAsMinimum` is true, upscaling is allowed. If false, `scaleDown()` is used to restrict scaling to only allow reducing the image size.